src/libarena/lib.rs

```diff
@@ -431,46 +392,41 @@ impl<T> TypedArenaChunk<T> {
         if intrinsics::needs_drop::<T>() {
             let mut start = self.start();
             for _ in 0..len {
-                ptr::read(start as *const T); // run the destructor on the pointer
-                start = start.offset(mem::size_of::<T>() as isize)
+                ptr::read(start); // run the destructor on the pointer
```

Comment:
There's a lot of stale comments in this file (from before this PR), for example, this destroy mentions type descriptors, but that's not relevant here, right?